ghga_connector/core/file_operations.py

```diff
@@ -70,15 +83,15 @@ def calc_part_ranges(
     ]
 
     if (total_file_size % part_size) > 0:
-        # if the last part is smaller than the part_size, calculate it range separately:
+        # if the last part is smaller than the part_size, calculate its range separately:
         part_ranges.append((part_size * full_part_number, total_file_size - 1))
 
     return part_ranges
 
 
 def download_file_parts(
     *,
-    download_url: str,
+    download_urls: Iterator[Union[Tuple[str, int, None], Tuple[None, None, int]]],
```

Comment:
That's the other way round as above: `Iterator[Union[Tuple[None, None, int], Tuple[str, int, None]]]`